docs/single-sourced-plugins.md

```diff
@@ -0,0 +1,82 @@
+# Single Sourced Plugins
+
+We use a Jekyll plugin (`plugin_single_source_generator.rb`) to dynamically generate pages from a single source file. It works as follows:
+
+- Read all navigation files (`app/_data/extensions/**/*.yml`)
+- For each version listed in the file, check if `app/_hub/[vendor]/[name]/[version].md` exists
+- If not, read `app/_hub/[vendor]/[name]/_index.md` and generate a version of the page
+- The latest version is always generated as `index.html`, whilst older versions are generated as `[version].html`
+
+## Concepts
```

Comment:
Was this heading left by accident, or are the following two headings supposed to be h3?

Reply:
Left in by accident, removing now